monkey/monkey_island/cc/ui/src/components/utils/SafeOptionValidator.js

```diff
@@ -18,29 +13,63 @@ function getPluginDescriptors(schema, config) {
   ]);
 }
 
+function getPluginDescriptors(schema, config) {
+  return ([
+    {
+      name: 'Exploiters',
+      allPlugins: schema.properties.propagation.properties.exploitation.properties.exploiters.properties,
+      selectedPlugins: Object.keys(config.propagation.exploitation.exploiters)
+    }
+  ])
+}
+
 function isUnsafeOptionSelected(schema, config) {
-  let pluginDescriptors = getPluginDescriptors(schema, config);
+  return isUnsafeLegacyPluginEnabled(schema, config) || isUnsafePluginEnabled(schema, config);
+}
+
+function isUnsafeLegacyPluginEnabled(schema, config) {
+  let pluginDescriptors = getLegacyPluginDescriptors(schema, config);
 
   for (let descriptor of pluginDescriptors) {
-    if (isUnsafePluginSelected(descriptor)) {
+    if (getUnsafeLegacyPlugins(descriptor).length > 0) {
       return true;
     }
   }
 
   return false;
 }
 
-function isUnsafePluginSelected(pluginDescriptor) {
-  let pluginSafety = new Map();
-  pluginDescriptor.allPlugins.forEach(i => pluginSafety[i.enum[0]] = i.safe);
+function getUnsafeLegacyPlugins(pluginDescriptor) {
```

Comment:
Idealy we should display which options are unsafe, because our warning is not actionable. Having a list should make it easier